src/libopensc/card-idprime.c

```diff
@@ -198,30 +486,56 @@ static int idprime_process_index(sc_card_t *card, idprime_private_data_t *priv,
 		/* in minidriver, mscp/kxcNN or kscNN lists certificates */
 		if (((memcmp(&start[4], "ksc", 3) == 0) || memcmp(&start[4], "kxc", 3) == 0)
 			&& (memcmp(&start[12], "mscp", 5) == 0)) {
+			int cert_id = 0;
+			idprime_container_t *container = NULL;
+
+			if (start[8] >= '0' && start[8] <= '9') {
```

Comment:
should we check the bounds of `start[7]` too here?

Reply:
I will fix this, but it should not be a problem as `kscXX` and `kxcXX` (hopefully) should have two-digit number index.